xds/internal/balancer/edsbalancer/balancergroup.go

```diff
@@ -295,8 +295,8 @@ func buildPickerAndState(m map[internal.Locality]*pickerState) (connectivity.Sta
 	return aggregatedState, newPickerGroup(readyPickerWithWeights)
 }
 
-// RandomWRR constructor, to be modified in tests.
-var newRandomWRR = wrr.NewRandom
+// WRR constructor, to be modified in tests.
+var newWRR = wrr.NewEDF
```

Comment:
RandomWRR is the behavior agreed on by all implementations of different gRPC languages. We cannot change the default.
We can make it configurable so you can override it in your client.

Changing the algorithm will also be easier in the future as there are also plans to support different locality picking algorithms.

Reply:
Was this discussion public? It looks a bit strange that gRPC uses different algorithm compared to Envoy (which uses EDF for this exact scenario)

Reply:
The xds gRFC is not public yet.

Is there any specific reason you care so much about this picking algorithm?

Reply:
If we decide to switch to EDF to be consistent with envoy, we should do that in all languages. I will bring this up with them.

Reply:
We care for endpoint lb (as edf provides fair distribution of the load across endpoints as in opposite to ≈random one).
For cross-zone lb we don't care that much (as number of zones is not that big) and the main/only motivator is consistency with other data plane (envoy).